feat: better explain signing (#183)
KILTprotocol/docs: docs/develop/01_sdk/02_cookbook/01_dids/06_full_did_tx.md

---
id: full-did-batch
title: Build DID Extrinsics
---

import TsJsBlock from '@site/src/components/TsJsBlock';

import FullDidSignTx from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/did/08_full_did_sign_tx.ts';
import FullDidBatch from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/did/08_full_did_batch.ts';

> A DID can be used to sign extrinsics.

Reviewer: ```suggestion
DID keys can be used to sign extrinsic.
```

> But not every extrinsics can be sign using a DID.

Reviewer: ```suggestion
But not every extrinsic can be signed using a DID.
```

The Spiritnet blockchain offers two flavours of extrinsics.

The first flavour can only be called using an account.

> These extrinsic require fee payment and need therefore be send from an account with sufficient balance.

Reviewer: ```suggestion
These extrinsic require fee payment and need therefore be sent from an account with sufficient balance.
```

Reviewer 2: It sounds like the second flavour of extrinsics do not involve any fee payment. I would probably switch them around, mentioning that there are some operations that require a DID signature. Everything else is the same. Both involve fee payment and regular account signatures.

Author: But we say in the second flavor that these need to be wrapped using the first flavor inorder for fees to be paid. So i think it's clear that both need to be paid.

Also we don't have terminology for these two types. We should give them names so that we can distinguish them. Account Extrinsic and DID Extrinsic?

Reviewer 2: At least I had this doubt when going over the text, I think at least the text can be reworded to explain the concept a bit differently.
I am ok with the Account vs DID Extrinsic distinction.

Author: I tried to make it clearer. [e16cf85](https://github.com/KILTprotocol/docs/pull/183/commits/e16cf85b28773e1eb24a03ee8b74d28ee4a30b4c)

> The second flavour is callable from a DID.

Reviewer: ```suggestion
The second callable flavour is from a DID.
```

Since every extrinsic requires fees to be paid, this flavour needs to be wrapped inside a normal account extrinsic.

> The account will pay all fees and deposit that will be required.

Reviewer: ```suggestion
All required fees and deposits will be paid by the account.
```

This document describes how to sign the DID extrinsics.
The KILT SDK provides two functions for signing DID extrinsics.
The first function signs a single extrinsic while the second one batches multiple extrinsics together.

## Single extrinsics

> To build a single extrinsic, you need to provide:

Reviewer 2: ```suggestion
To sign a single extrinsic, you need to provide:
```

> * the DID that should be the origin of the extrinsic

Reviewer 2: ```suggestion
* the DID that wants to sign the extrinsic (also called *origin* of the extrinsic)
```

Reviewer 2: Maybe the order of these elements can match the order of the parameters in the function signature? That would make it a bit easier for the users to follow the two at the same time.

* a [`SignCallback`](../06_signCallback.md) that signs the extrinsic

> * the extrinsic that should be submitted

Reviewer 2: ```suggestion
* the extrinsic that should be signed and submitted
```
